clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferIndex.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.clio.transfer.model
+
+import io.circe.{Decoder, Encoder, Json}
+import org.broadinstitute.clio.util.json.{JsonSchemas, ModelAutoDerivation}
+
+/**
+  * This models an index in our database. An index is a table (schema) within our database.
+  */
+sealed trait TransferIndex extends ModelAutoDerivation {
+
+  /**
+    * Encode the index as a url segment for making web service API calls
+    *
+    * @return the url path representation of this index
+    */
+  def urlSegment: String
+
+  def jsonSchema: Json
+
+  def name: String
+
+  def commandName: String
+
+  type metadataType
```

Comment:
Nitpick: even for abstract type members, I think the convention is to use upper camel case.

Reply:
OK